lib/resolve-shims.js

```diff
@@ -155,16 +156,18 @@ function resolveInlineShims(packageDir, pack, shimField, messages) {
   return { shims: resolvedShims, exposeGlobals: separated.exposeGlobals };
 }
 
-var resolve = module.exports = function resolveShims (file, messages, cb) {
+var resolve = module.exports = function resolveShims (file, messages, shimConfig, cb) {
   // find the package.json that defines browserify-shim config for this file
-  mothership(file, function (pack) { return !! pack['browserify-shim'] }, function (err, res) {
+  // make package.json config optional when transform options are provided
+  mothership(file, function (pack) { return !! shimConfig || !! pack['browserify-shim'] }, function (err, res) {
```

Comment:
This isn't correct -- putting this line here will make mothership stop looking after the first step up in the filesystem. This needs to be handled on L165.